layers/utils/segment_get_urls.py

```diff
@@ -0,0 +1,236 @@
+
+tracer = Tracer()
+
+
+@tracer.capture_method(capture_response=False)
+def create_presigned_get_url(s3_url: str) -> Tuple[str, str]:
+    """Generate a presigned URL for S3 object access.
+
+    Args:
+        s3_url: The original S3 URL
+
+    Returns:
+        Tuple of (original_url, presigned_url)
+    """
+    url_parse = urlparse(s3_url)
+    url = generate_presigned_url(
+        "get_object", url_parse.netloc.split(".")[0], url_parse.path.split("/", 1)[1]
+    )
+    return (s3_url, url)
+
+
+@tracer.capture_method(capture_response=False)
+def create_direct_s3_get_url(
+    object_id: str, storage_backend: Dict[str, str], include_storage_id: bool
+) -> Dict[str, str]:
+    """Generate a non-signed S3 URL for object access.
+
+    Args:
+        object_id: The S3 object identifier
+        storage_backend: Storage backend configuration dict
+
+    Returns:
+        Dict containing label and URL
+    """
+    get_url = {
+        "label": storage_backend["label"],
+        "url": f'https://{storage_backend["bucket_name"]}.s3.{storage_backend["region"]}.amazonaws.com/{object_id}',
+    }
+    if include_storage_id:
+        get_url["storage_id"] = storage_backend["id"]
+    return get_url
+
+
+@tracer.capture_method(capture_response=False)
+def create_presigned_urls_parallel(url_set: Set[str]) -> Dict[str, str]:
+    """Generate presigned URLs in parallel using ThreadPoolExecutor.
+
+    Args:
+        urls: Set of s3 URLs to generate presigned URLs for
+
+    Returns:
+        Dict mapping original URLs to presigned URLs
+    """
+    # Asynchronous call to pre-signed url API
+    with concurrent.futures.ThreadPoolExecutor() as executor:
+        futures = []
+        for s3_url in url_set:
+            futures.append(executor.submit(create_presigned_get_url, s3_url))
+    # Build dict of returned urls
+    url_mapping = {}
+    for future in futures:
+        key, url = future.result()
+        url_mapping[key] = url
+    return url_mapping
+
+
+@tracer.capture_method(capture_response=False)
+def get_storage_backends(
+    accept_storage_ids: Optional[str],
+    segments: List[Dict],
+    default_storage_backend_id: str,
+) -> Dict[str, Dict]:
+    """Get storage backend configurations for segments.
+
+    Args:
+        accept_storage_ids: Comma-separated storage IDs to filter by
+        segments: List of segment dictionaries
+        default_storage_backend_id: The id of the default storage backend
+
+    Returns:
+        Dict mapping storage IDs to their backend configurations
+    """
+    filter_ids = accept_storage_ids.split(",") if accept_storage_ids else None
+    distinct_storage_ids = set(
+        storage_id
+        for segment in segments
+        for storage_id in segment.get("storage_ids", [default_storage_backend_id])
+    )
+    filtered_storage_ids = (
+        [storage_id for storage_id in distinct_storage_ids if storage_id in filter_ids]
+        if filter_ids
+        else distinct_storage_ids
+    )
```

Comment:
```suggestion
    filtered_storage_ids = distinct_storage_ids & filter_ids if filter_ids else distinct_storage_ids
```

You can save yourself the verbiage of list comprehension by using the intersection of two sets.